enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,316 @@
+
+The backup sidecar container is being `disabled` by default. Upon creating a `Backup` CR on an Openshift cluster, the `PeriodicBackupController` reacts by enabling the container. This leads to a change in etcd static pod manifest, and a new revision is being rolled out. As explained above, such action causes etcd-cluster and API-Server unavailability.
+
+Additionally, any error within the backup sidecar, the container goes into `CrashLoopBackOff`, which leads to the whole `Pod` reports `NotReady` status. Consequently, the whole `Etcd-cluster` can not server requests, nor the `API-Server`.
+
+
+#### DaemonSet manages an Etcd-backup pod within every control-plane node.
+
+This approach separates the `etcd-backup-server` deployment from the etcd static pods, to avoid affecting etcd-cluster and API-Server availability upon any error within the etcd-backup-server.
+
+The advantage of using `DaemonSet` over Deployment is to guarantee scheduling etcd-backup-server pods on all `Control-plane` nodes, regardless of the number of nodes (e.g. 1, 3, 5) using `nodeSelector` capability.
+By setting the `nodeSelector` to the master node label (i.e. `node-role.kubernetes.io/master`), etcd-backup-server's pods are guaranteed to scheduled on every control-plane node.
+
+This method has been implemented and tested. The same container used in the Sidecar approach has been utilised. However, the scheduling is being managed by a DaemonSet.
+Upon creating a `Backup` CR on an Openshift cluster, the `PeriodicBackupController` reacts by creating a DaemonSet resource, with the `etcd-backup-server` container and the configurations applied in the CR.
+During [testing](https://github.com/openshift/cluster-etcd-operator/pull/1354#issuecomment-2408661261), this method has shown a solid and stable deployment, where the Backup pods shown only `1` restart over a duration of `45` minutes. Moreover, the cause of the restart on all the Backup pods were due to `node` issue, not the backup container.
+
+_**Pros**_
+
+As the Backup pods are being scheduled independently of the etcd pods, any error within the Backup pods does not influence cluster Openshift cluster availability.
+
+Moreover, as the Backup pods are being scheduled within every master nodes, the backups are being taken from every local etcd member on the same node. This avoids overwhelming a specific member with backup requests.
+
+Since a Backup pod exist on every master node, and a `hostPath` volume is being used to store the backups, it is guaranteed to have available backups on all master nodes, which is required for disaster recovery scenarios.
+
+_**Cons**_
+
+No issue has been encountered during testing this approach.
+
+### Open Questions
+
+- What values could be used for default configurations, considering all OCP variants.
```

Comment:
Not sure which "values" this is about - requests/limits?

Reply:
How about a default interval such that every node has a backup from the last 24h and retention being three backups per node in case there's a need to find a backup from before a particular change in the last 72h.

Reply:
I actually used this `00:00` as default value 

See https://github.com/openshift/cluster-etcd-operator/pull/1354/files#r1882979478 

here is the [commit](https://github.com/openshift/cluster-etcd-operator/pull/1354/commits/564eb9ac3667ae9801c9276b3ac251f1d243be2b) fixing this [bug](https://issues.redhat.com/browse/OCPBUGS-43687) as reported by @sandeepknd

Reply:
need consensus from everyone 

cc @openshift/openshift-team-etcd  @vrutkovs @JoelSpeed

Reply:
David's suggestion seems reasonable.

Did we conclude on the thread about taking an initial backup when the node is first created? (to avoid CPMS rollout clearing all backups and then needing a backup between the rollout and 00:00)

Reply:
I think it is best to discuss during the control plane meeting, so we get input from everyone 

Wdyt @JoelSpeed @deads2k @tjungblu  ?

Reply:
> Did we conclude on the thread about taking an initial backup when the node is first created? (to avoid CPMS rollout clearing all backups and then needing a backup between the rollout and 00:00)

I think we can add this to GA requirement, wdyt @deads2k

Reply:
> iirc, the new nodes created by CPMSO does not become running till the etcd pod on the node not only joins the etcd cluster, but also becomes voting member

This is not true, CEO and etcd have nothing to do with the Node becoming ready, or the Machine moving into the Running phase.

The CEO adds a machine deletion hook to the machine, when it intends to promote an etcd member to voting. This is part of the machine spec, and so you couldn't gate scheduling on it using normal scheduler mechanics.

Reply:
Ok, what do you think if CEO labels machines according to its role in etcd cluster (i.e. learner, voting) ?

Reply:
If CEO were to add a label to the nodes, which stated whether they were voting or learner members, then yes, that could be used as a scheduling gate to make sure backups aren't taken until members are voting.

Another part of this thread, which I think we need to discuss wider, is whether we are concerned about CPMS rollouts and this mechanism.

Eg, if we are backup up daily at midnight, and I rollout CPMS at around 8am, there's a potentially long period where there will be no backups (CPMS deleting last old machine to midnight). And are we concerned about this scenario?

We could argue that for CPMS users they should use an external backup, and this mechanism is best effort, but lets make sure we have stakeholder agreement on that.

The alternative is to try and come up with a way to schedule backups each time a member is promoted, which would mean we _should_ always have at least 1 backup

Reply:
> How about a default interval such that every node has a backup from the last 24h and retention being three backups per node in case there's a need to find a backup from before a particular change in the last 72h.

I'm ok with this, but we should get agreement about what happens when CPMS wipes this state. Perhaps discussing on the control plane arch on tuesday is worthwhile?

Reply:
@JoelSpeed  I raised this [commit](https://github.com/openshift/cluster-etcd-operator/pull/1354/commits/7706a916815d563bdb1ba4c036e53d3c462a513b) .. Here is the changes 

- List All Etcd Voting Members
- List All Master Nodes
- Use All Master Node with Etcd Voting Member
- Apply a `votingNode` label to all of them to be used by the DS for scheduling
- Upon removing the `default` Backup CR, delete `votingNode` label from all master nodes


creating a cluster to see it in action now

Reply:
CEO should not touch the node object, if anything we would keep it in the backup CRD status somehow